### PR TITLE
Custom GraphQL and Mobility Profile Routing

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR sets up craco to accept custom plan GraphQL files (or use a default one).
Custom plan query files are passed through the custom `config.js` and placed in the redux state at startup.
For configuration-enabled mobility profiles, that new parameter is added to the variables passed to the GraphQL `plan` query. 

To test, ensure that planning trips with mobility profiles produce different results based on the profiles, and that on configs without mobility profiles, planning trips is not affected.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by josh-willis-arcadis — APPROVED** [review]
changes are working with multiple configs with and without mobility profile.

File: lib/main.js [hunk 1]
```diff
@@ -34,6 +34,18 @@ import(CSS)
...
+// defaults to ./config.js
+// defined in webpack config:
+// The JS_CONFIG variable is passed to this file by webpack's `DefinePlugin` that replaces the variable
+// with its content at compile time (like C's `#define` preprocessor directive).
+// eslint-disable-next-line no-undef
+const jsConfig = require(JS_CONFIG).configure(otpConfig)
+
+// Plug the plan query into the config (if available)
+otpConfig.api.planQuery = jsConfig.planQuery
+
```
Comment (miles-grant-ibigroup) [inline lib/main.js]: Do we still need this? Why not just put it in the js config?

File: craco.config.js [hunk 2]
```diff
@@ -65,6 +67,13 @@ module.exports = {
       }
       addBeforeLoader(webpackConfig, loaderByName('file-loader'), yamlLoader)
 
+      // Support YAML
```
Comment (miles-grant-ibigroup) [inline craco.config.js]: Comment needs updating

File: lib/actions/apiV2.js [hunk 3]
```diff
@@ -1046,6 +1048,9 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
       ...currentQuery,
       numItineraries: numItineraries || getDefaultNumItineraries(config)
     }
+    if (config.mobilityProfile) {
+      baseQuery.mobilityProfile = loggedInUser?.mobilityProfile?.mobilityMode
+    }
```
Comment (miles-grant-ibigroup) [inline lib/actions/apiV2.js:1054]: Do we have a GMAP specific branch for otp-rr? Do we want to bring this into dev?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Overall works, even if it's a bit hacky. Just want to double check a few things!

File: lib/actions/apiV2.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/actions/apiV2.js:1054]: Not yet, although try to move specific stuff to config would be best.

File: lib/main.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/main.js]: This puts the custom query into the redux state, so it can be retrieved from actions.

File: craco.config.js (hunk 2 above)
Comment (binh-dam-ibigroup) [reply craco.config.js]: Thanks, updated in https://github.com/opentripplanner/otp-react-redux/pull/1292/commits/37260077aab004d45b0ced9dbef13c10226dbb8c

File: lib/main.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/main.js]: I updated the comment in https://github.com/opentripplanner/otp-react-redux/pull/1292/commits/e60f6942111a56e4be995cdb0349a5522f30d94a.

**Review by miles-grant-ibigroup — APPROVED** [review]
A bit nervous about this but it seems to work alright...

**Pull request merged**